src/app.py

```diff
@@ -508,59 +624,40 @@ def get_entity_by_id(id):
+    if entity_scope == DataVisibilityEnum.PUBLIC:
+        user_authorized = True
     else:
-        # The `data_access_level` of Donor/Sample can only be either 'public' or 'consortium'
-        if entity_dict['data_access_level'] == ACCESS_LEVEL_CONSORTIUM:
-            token = get_user_token(request, non_public_access_required = True)
-
-    # By now, either the entity is public accessible or the user token has the correct access level
-    # We'll need to return all the properties including those
-    # generated by `on_read_trigger` to have a complete result
-    # E.g., the 'next_revision_uuid' and 'previous_revision_uuid' being used below
-    # On entity retrieval, the 'on_read_trigger' doesn't really need a token
-    complete_dict = schema_manager.get_complete_entity_result(token, entity_dict)
+        # It's highly possible that there's no token provided
+        user_token = get_user_token(request)
 
-    # Additional handlings on dataset revisions
-    # The rule is that a revision can only be made against a published dataset,
-    # so it should never occur that a consortium level revision is between two published revisions
-    # However, the very last dataset revision can be non-published
-    if schema_manager.entity_type_instanceof(normalized_entity_type, 'Dataset'):
-        # The `next_revision_uuid` is only availabe in complete_dict because it's generated by the 'on_read_trigger'
-        property_to_pop = 'next_revision_uuid'
-
-        # When the dataset is published but:
-        # - There's no Authorization header thus no token
-        # - The groups token is valid but the user doesn't belong to HuBMAP-READ group
-        # - Or the token is valid but doesn't contain group information (auth token or transfer token)
-        # We need to remove the `next_revision_uuid` from response
-        # Otherwise, we should send back the `next_revision_uuid` (if exists) when the member belongs to HuBMAP-Read group
-        if entity_dict['status'].lower() == DATASET_STATUS_PUBLISHED:
-            if not user_in_hubmap_read_group(request):
-                if property_to_pop in complete_dict:
-                    revision_entity_dict = query_target_entity(complete_dict[property_to_pop], token)
-                    # Remove the property from the resulting complete_dict
-                    # if the revision is not published
-                    if revision_entity_dict['status'].lower() != DATASET_STATUS_PUBLISHED:
-                        complete_dict.pop(property_to_pop)
-        # Non-published dataset can NOT have next revisions
+        # The user_token is flask.Response on error
+        # Without token, the user can only access public collections, modify the collection result
+        # by only returning public datasets attached to this collection
+        if isinstance(user_token, Response):
+            forbidden_error(f"{normalized_entity_type} for {id} is not accessible with credentials presented.")
```

Comment:
Change "with" to "without"

Reply:
This message is specific to no credentials at all being presented (and does not cover bad credentials.)  Change message to `f"{normalized_entity_type} for {id} is not accessible without presenting a token."`